src/Events/Integrations/Modules/Tickets_Emails/Email/RSVP.php

```diff
@@ -133,47 +133,78 @@ public function include_attachments( $attachments, $email_id, $email_class ) {
 	 * @return void
 	 */
 	public function include_event_links( $parent_template ) {
-		$email_class = tribe( RSVP_Email::class );
+		if ( ! $this->should_show_links( $parent_template, $email_class, $args ) ) {
+			return;
+		}
 
-		// Bail early if the email class is not enabled.
-		if ( ! $email_class->is_enabled() ) {
+		$args['event_gcal_link'] = tribe( Google_Calendar::class )->generate_single_url( $args['event']->ID );
+		$args['event_ical_link'] = tribe_get_single_ical_link( $args['event']->ID );
+
+		if ( ! empty( $args['preview'] ) ) {
+			$args['event_gcal_link'] = '#';
+			$args['event_ical_link'] = '#';
+		}
+
+		tribe( Template::class )->template( 'template-parts/body/event/links', $args, true );
+	}
+
+	/**
+	 * Maybe include event link styles.
+	 *
+	 * @since TBD
+	 *
+	 * @param \Tribe__Template $parent_template Event Tickets template object.
+	 *
+	 * @return void
+	 */
+	public function include_event_link_styles( $parent_template ): void {
+		if ( ! $this->should_show_links( $parent_template, $email_class, $args ) ) {
 			return;
 		}
 
-		$use_ticket_email = tribe_get_option( $email_class->get_option_key( 'use-ticket-email' ), false );
+		tribe( Template::class )->template( 'template-parts/header/head/tec-styles', $args, true );
+	}
 
+	/**
+	 * Determines whether or not RSVP should show calendar links.
+	 *
+	 * @since TBD
+	 *
+	 * @param \Tribe__Template    $parent_template Event Tickets template object.
+	 * @param Email_Abstract|null $email_class     References email class.
+	 * @param array|null          $args            References template context arguments.
+	 *
+	 * @return bool
+	 */
+	public function should_show_links( $parent_template, &$email_class, &$args ): bool {
```

Comment:
Still confused to why are we passing as reference?

Reply:
The `$email_class` and `$args` are needed within the `should_show_links()` method and within the calling method. Rather than creating these variables in both methods, I referenced them. In some cases, the `$args` variable doesn't need to be created, so I didn't want to create them both, then pass them into the method.

Reply:
That is correct an sound logic, the only problem is with the maintenance of that code. If someone makes changes to the reference it could create fatals that are unwanted. Does it work without passing by reference?

Reply:
Got it changed. Thank for being a bouncing board!